lib/filterx/expr-literal-generator.c

```diff
@@ -150,6 +150,22 @@ filterx_literal_dict_generator_foreach(FilterXExpr *s, FilterXLiteralDictGenerat
   return TRUE;
 }
 
+gboolean
+filterx_literal_list_generator_foreach(FilterXExpr *s, FilterXLiteralListGeneratorForeachFunc func, gpointer user_data)
+{
+  FilterXExprLiteralGenerator *self = (FilterXExprLiteralGenerator *) s;
+
+  for (GList *link = self->elements; link; link = link->next)
+    {
+      FilterXLiteralGeneratorElem *elem = (FilterXLiteralGeneratorElem *) link->data;
+
+      if (!func(elem->value, user_data))
```

Comment:
Can we also expose the `index` of the element as well to the user supplied function?

Reply:
Good idea.